eLearning-courseware/eLearning.jsonld

```diff
@@ -0,0 +1,1271 @@
+        },
+        "definition": {
+          "en": "Learner has initialized the course."
+        },
+        "rules": [
+          {
+            "location": "$.actor.mbox",
+            "presence": "included"
+          },
+          {
+            "location": "$.object.id",
+            "presence": "included"
+          },
+          {
+            "location": "$.object.definition.name",
+            "presence": "included"
+          },
+          {
+            "location": "$.object.definition.description",
+            "presence": "included"
+          },
+          {
+            "location": "$.verb.id",
+            "any": ["http://adlnet.gov/expapi/verbs/initialized"]
+          }
+        ]
+      },
+      {
+        "id": "https://github.com/TryxAPI/xAPI-eLearning-Courseware-Profile/templates/courseLaunched",
+        "type": "StatementTemplate",
+        "inScheme": "https://w3id.org/xapi/eLearning",
+        "prefLabel": {
+          "en": "Course launched"
+        },
+        "definition": {
+          "en": "Learner has launched the course."
+        },
+        "rules": [
+          {
+            "location": "$.actor.mbox",
+            "presence": "included"
+          },
+          {
+            "location": "$.verb.id",
+            "any": ["http://adlnet.gov/expapi/verbs/launched"]
+          },
+          {
+            "location": "$.object.id",
+            "presence": "included"
+          },
+          {
+            "location": "$.object.definition.name",
+            "presence": "included"
+          },
+          {
+            "location": "$.object.definition.description",
+            "presence": "included"
+          },
+          {
+            "location": "$.object.definition.type",
+            "any": ["http://adlnet.gov/expapi/activities/course"]
+          }
+        ]
+      },
+      {
+        "id": "https://github.com/TryxAPI/xAPI-eLearning-Courseware-Profile/templates/pageCompleted",
+        "type": "StatementTemplate",
+        "inScheme": "https://w3id.org/xapi/eLearning",
+        "prefLabel": {
+          "en": "Page completed"
+        },
+        "definition": {
+          "en": "Learner has completed the page."
+        },
+        "rules": [
+          {
+            "location": "$.actor.mbox",
+            "presence": "included"
+          },
+          {
+            "location": "$.verb.id",
+            "any": ["http://adlnet.gov/expapi/verbs/completed"]
+          },
+          {
+            "location": "$.object.id",
+            "presence": "included"
+          },
+          {
+            "location": "$.object.definition.name",
+            "presence": "included"
+          },
+          {
+            "location": "$.object.definition.description",
+            "presence": "included"
+          },
+          {
+            "location": "$.result.completion",
+            "any": [true]
+          },
+          {
+            "location": "$.context.extensions.pageId",
```

Comment:
The key wouldn't be pageId, but some extension URI. You can easily get that into JSONPath by writing: "$.context.extensions['http://example.com/extension/uri']"

Reply:
The pageId and topicId rules were removed from all the templates. This will be included in an upcoming revision. We need to write the rules for the extension first.